enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2054,6 +2082,50 @@ func (ws *workspace) fetch(ctx context.Context, remoteURL string, refs []string,
 	return nil
 }
 
+// Writes a wrapper script that invokes the ci_runner with the bazel_wrapper subcommand.
+// Also adds it to the PATH so it will be invoked whenever `bazel` or `bazelisk` are called.
+// The wrapper script adds a startup option for the custom ci_runner .bazelrc to
+// all bazel commands.
+func (ws *workspace) writeBazelWrapperScript() error {
+	wrapperDir := filepath.Join(ws.rootDir, "wrappers")
+	for _, c := range []string{bazelBinaryName, bazeliskBinaryName} {
+		wrapperPath := filepath.Join(wrapperDir, c)
+		_, err := os.Stat(wrapperPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				if err := os.MkdirAll(filepath.Dir(wrapperPath), 0755); err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+
+		f, err := os.OpenFile(wrapperPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0755)
+		if err != nil {
+			return status.InternalErrorf("Failed to open %s for writing: %s", wrapperPath, err)
+		}
+		defer f.Close()
+
+		contents := `
+#!/usr/bin/env sh
+BAZEL_WRAPPER_MODE=1 BAZEL_BIN=` + *bazelCommand + ` CI_RUNNER_ROOT=` + ws.rootDir + ` ` + os.Getenv("BUILDBUDDY_CI_RUNNER_ABSPATH") + ` "$@"
```

Comment:
If I'm understanding correctly, what you're talking about is handled here: https://github.com/buildbuddy-io/buildbuddy/pull/6961/files#diff-a1778f788b2996ecb914bb66cf733fe40e53a23e1d348e392d9e19db1f1b8d77R2568

We want to replace the wrapper script with the bazel process. I don't think we also need to replace the process that's calling the wrapper script, right?

Reply:
The process tree looks like this:

```
bash (user script)
  > sh (this `bazel` script)
    > `bazel_wrapper` (go process)
      > `bazelisk`
```

With your `syscall.Exec()` in `bazel_wrapper`, that makes the process tree look like this (after `bazelisk` is started):

```
bash (user script)
  > sh (this `bazel` script)
    > `bazelisk`
```

To remove the remaining `sh` process from the tree, we need another `exec` here

Reply:
I see. Updated the command to use `exec`, but I didn't include `env -` because we still want to maintain the env vars set for the parent process. Earlier in the ci_runner we set things like BUILDBUDDY_ARTIFACTS_DIRECTORY and USER